src/Sets/HParallelotope.jl

```diff
@@ -16,57 +16,65 @@ Type that represents a parallelotope in constraint form.
 ```
-for ``i = 1, …, n``. Here ``D_i`` represents the ``i``-th row of ``D`` and ``c_i`` the ``i``-th
-component of ``c``.
+for ``i = 1, …, n``. Here ``D_i`` represents the ``i``-th row of ``D`` and
+``c_i`` the ``i``-th component of ``c``.
 
-For details on the notions given in these notes as well as applications of parallelotopes in
-reachability analysis we refer to [1] and [2]. For conversions between set representations
-we refer to [3].
+Note that, although representing a zonotopic set, an `HParallelotope` can be
+empty if the constraints are contradictory. This may cause problems with default
+methods because the library assumes that zonotopic sets are non-empty. Thus such
+instances are considered illegal.
+
+For details as well as applications of parallelotopes in reachability analysis
+we refer to [1] and [2]. For conversions between set representations we refer to
+[3].
 
 ### References
 
-[1] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Reachability computation for polynomial dynamical systems.*
-    Formal Methods in System Design 50.1 (2017): 1-38.
+[1] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Reachability computation for
+polynomial dynamical systems.* Formal Methods in System Design 50.1 (2017): 1-38.
 
-[2] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Parallelotope bundles for polynomial reachability.*
-    Proceedings of the 19th International Conference on Hybrid Systems: Computation and Control. ACM, 2016.
+[2] Tommaso Dreossi, Thao Dang, and Carla Piazza. *Parallelotope bundles for
+polynomial reachability.* Proceedings of the 19th International Conference on
+Hybrid Systems: Computation and Control. ACM, 2016.
 
-[3] Matthias Althoff, Olaf Stursberg, and Martin Buss. *Computing reachable sets of hybrid systems using
-    a combination of zonotopes and polytopes.* Nonlinear analysis: hybrid systems 4.2 (2010): 233-249.
+[3] Matthias Althoff, Olaf Stursberg, and Martin Buss. *Computing reachable sets
+of hybrid systems using a combination of zonotopes and polytopes.* Nonlinear
+analysis: hybrid systems 4.2 (2010): 233-249.
 """
 struct HParallelotope{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
     directions::MN
     offset::VN
 
     # default constructor with dimension check
     function HParallelotope(D::MN, c::VN) where {N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
-        @assert length(c) == 2*checksquare(D) "the length of the offset direction should be twice the size " *
-            "of the directions matrix, but they are $(length(c)) and $(size(D)) dimensional respectively"
+        @assert length(c) == 2*checksquare(D) "the length of the offset " *
+            "vector should be twice the size of the directions matrix, " *
+            "but they have sizes $(length(c)) and $(size(D)) respectively"
         return new{N, VN, MN}(D, c)
     end
 end
 
 isoperationtype(::Type{<:HParallelotope}) = false
-isconvextype(::Type{<:HParallelotope}) = true
```

Comment:
why is this removed? is it because it'll use some interface method?

Reply:
Yes, I removed this from all concrete types so far if their supertype already defined it the same way.